cpp/src/parquet/statistics.cc

```diff
@@ -621,7 +628,9 @@ class TypedStatisticsImpl : public TypedStatistics<DType> {
       // num_values_ is reliable and it means number of non-null values.
       s.all_null_value = num_values_ == 0;
     }
-    // TODO (GH-36505): distinct count is not encoded for now.
+    if (HasDistinctCount()) {
```

Comment:
@mapleFU FYI I noticed that distinct counts aren't propagated to `EncodedStatistics`, as tracked in https://github.com/apache/arrow/issues/36505. For my new test, I could potentially just check `TypedStatistics` without calling `Encode()`, if you'd rather that I leave this as-is.

Reply:
Yes, previously I think we can write it, but pirtou mentioned that it's bug-prone, and other implemention didn't write that. So I set it not encoded.

Reply:
Sorry, by, "That seems ok to me," do you mean to leave this here? Or to remove it?